src/api/routes/controllers/chats.controller.js

```diff
@@ -0,0 +1,68 @@
+const jwt = require("jsonwebtoken");
+const chatsService = require("../../services/chats.service");
+
+exports.getChat = async (req, res, next) => {
+  const { id } = req.params;
+  const { lastIndex } = res.locals;
+  const DEFAULT_LENGTH = 30;
+
+  try {
+    const response = await chatsService.getChatsFromGoalId(id);
+
+    if (!lastIndex) {
+      const rangeStart =
+        response.length < DEFAULT_LENGTH ? 0 : response.length - DEFAULT_LENGTH;
+      const chats = response.slice(rangeStart, response.length);
+      const newNextPageToken =
+        rangeStart === 0
+          ? null
+          : jwt.sign(
+              {
+                lastIndex: rangeStart,
+              },
+              process.env.CHAT_QUERY_SECRET_KEY,
+              {
+                expiresIn: "1d",
+              },
+            );
+
+      return res.json({
+        result: {
+          messages: chats,
+          nextPageToken: newNextPageToken,
+        },
+      });
+    }
+
+    const rangeStart =
+      lastIndex < DEFAULT_LENGTH ? 0 : lastIndex - DEFAULT_LENGTH;
+    const chats = response.slice(rangeStart, lastIndex);
+    const newNextPageToken =
+      rangeStart === 0
+        ? null
+        : jwt.sign(
+            {
+              lastIndex: rangeStart,
+            },
+            process.env.CHAT_QUERY_SECRET_KEY,
+            {
+              expiresIn: "1d",
+            },
+          );
+
+    return res.json({
+      result: {
+        messages: chats,
+        nextPageToken: newNextPageToken,
+      },
```

Comment:
위 아래 로직이 중복되는 로직이네요!
하나의 유틸함수로 만들어서 로직을 줄여보는 것도 좋을 것 같습니다👍

```suggestion
    function makeNewNextPageToken(parameter) {
      const rangeStart =
        parameter < DEFAULT_LENGTH ? 0 : parameter- DEFAULT_LENGTH;
      const chats = response.slice(rangeStart, parameter);
      const newNextPageToken =
        rangeStart === 0
          ? null
          : jwt.sign(
              {
                lastIndex: rangeStart,
              },
              process.env.CHAT_QUERY_SECRET_KEY,
              {
                expiresIn: "1d",
              },
            );

      return res.json({
        result: {
          messages: chats,
          nextPageToken: newNextPageToken,
        },
      });
    }

```
더 좋은 함수명과 파라미터명이 있을 듯 하네요

Reply:
활용도 측면에서 따로 유틸로 빼야 할까 고민이 되긴 하는데요, 중복 로직은 내부에 함수로 선언해서 조금 더 리팩토링해서 올리겠습니다. 🙂